src/cysignals/cysignals_config.h.in

```diff
@@ -28,3 +28,16 @@
 #define cysetjmp(env) setjmp(env)
 #define cylongjmp(env, val) longjmp(env, val)
 #endif
+
+#if !defined(__MINGW32__) && !defined(_WIN32)
+#define POSIX
+#else
+/*
+define unused signals for windows and give each a different value to compile
+when two or more a these signals are used in the same switch.
+*/
+#define SIGHUP -1
+#define SIGBUS -2
+#define SIGALRM -3
+#define SIGQUIT -4
```

Comment:
What's the deal with these signals? Are they defined on Windows or not? The issue with the switch statement has been fixed.

Reply:
They are not defined on Windows and have been added to reduce `#ifdef POSIX` usage. I will try to remove them.

Reply:
I didn't say that you have to remove this. I was just a bit confused with the comment referring to an issue with `switch` which is no longer relevant.

Reply:
I don't get it. Why is it no longer relevant ?  We still have a switch with `SIGHUP`, `SIGBUS` and `SIGQUIT` in `sigdie_for_sig` function.

Reply:
Anyway i have at least one type here` when two or more **a** these signals`

Reply:
>  We still have a switch with `SIGHUP`, `SIGBUS` and `SIGQUIT` in `sigdie_for_sig` function.

Right, there were two such `switch` statements and I fixed only one. I will fix the other right away.

Reply:
Ok. Do you prefer i revert all signal to -1 or adapting the comment is enough ?
Leaving it as it is also made sense because it remain true that having different values allow switch usage even if the current code doesn't use one.

Reply:
I think I prefer what you tried initially: mapping those signals to other existing signals. Something like
```
#define SIGHUP SIGTERM
#define SIGBUS SIGSEGV
#define SIGALRM SIGINT
#define SIGQUIT SIGTERM
```
That way, we still do `raise(SIGQUIT)` if we want.

Reply:
> I think I prefer what you tried initially: mapping those signals to other existing signals. Something like
> 
> ```
> #define SIGHUP SIGTERM
> #define SIGBUS SIGSEGV
> #define SIGALRM SIGINT
> #define SIGQUIT SIGTERM
> ```
> That way, we still do `raise(SIGQUIT)` if we want.

I see the idea but i think doing that can cause unwanted side effects or cause additional constraints for future developments. For example we currently have this code in sigdie_for_sig :
```
    else if (sig == SIGSEGV)
        sigdie(sig, "Unhandled SIGSEGV: A segmentation fault occurred.");
    else if (sig == SIGBUS)
        sigdie(sig, "Unhandled SIGBUS: A bus error occurred.");
```
That work fine with `#define SIGBUS SIGSEGV` but if SIGBUS is tested before SIGSEV, after some modification, you will get a wrong message on windows.

Reply:
> That work fine with `#define SIGBUS SIGSEGV` but if `SIGBUS` is tested before `SIGSEV`, after some modification, you will get a wrong message on windows.

Still, we could document there that the order is important. There are 6 standard signals according to the C standard (https://en.wikipedia.org/wiki/C_signal_handling#Standard_signals) and those are the ones that Windows supports. So it's simply a matter of putting the standard signals first.